src/openms/include/OpenMS/KERNEL/MSSpectrum.h

```diff
@@ -728,6 +731,12 @@ namespace OpenMS
 
       for (Size i = 0; i < float_data_arrays_.size(); ++i)
       {
+        if (float_data_arrays_[i].size() != peaks_old)
+        {
+          throw Exception::Precondition(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "FloatDataArray[" + String(i) + "] size (" + 
+            String(float_data_arrays_[i].size()) + ") does not match spectrum size (" + String(peaks_old) + ")");
+        }
```

Comment:
yes it is, we actually just recently had a discussion about it and added cv terms that are specifically designed for this:

https://sourceforge.net/p/psidev/mailman/psidev-ms-vocab/thread/a797e011d942e997f15d94c5edf55bcc%40mail.gmail.com/#msg35673878

so there are at least three terms that have "subsampled" data in them